x/magpie/internal/keeper/handler.go

```diff
@@ -162,33 +37,48 @@ func handleMsgCreateSession(ctx sdk.Context, keeper Keeper, msg types.MsgCreateS
 		sdk.NewEvent(
 			sdk.EventTypeMessage,
 			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+			sdk.NewAttribute(sdk.AttributeKeyAction, types.ActionCreationSession),
 			sdk.NewAttribute(sdk.AttributeKeySender, msg.Owner.String()),
 		),
 	)
 
-	pkBytes, _ := base64.StdEncoding.DecodeString(msg.Pubkey)
-
+	// Get the public key used to sign the message
+	pkBytes, _ := base64.StdEncoding.DecodeString(msg.PubKey)
 	var pkBytes33 = [33]byte{}
 	copy(pkBytes33[:], pkBytes)
 	pubkey := secp256k1.PubKeySecp256k1(pkBytes33)
 
-	message := fmt.Sprintf(`{"account_number":"0","chain_id":"%s","fee":{"amount":[],"gas":"200000"},"memo":"","msgs":[{"type":"desmos/MsgCreateSession","value":{"created":"%s","external_owner":"%s","namespace":"%s","owner":"%s","pubkey":"%s","signature":null}}],"sequence":"0"}`,
-		ctx.ChainID(), msg.Created.Format(time.RFC3339Nano), msg.ExternalOwner, msg.Namespace, msg.Owner.String(), msg.Pubkey)
+	// Create the StdSignDoc by using the given message data, with an empty string
+	signedMsg := msg
+	signedMsg.Signature = ""
+
+	stdSignDoc := auth.StdSignDoc{
+		AccountNumber: 0,
+		ChainID:       msg.Namespace,
+		Fee:           json.RawMessage(auth.NewStdFee(200000, nil).Bytes()),
+		Memo:          "",
+		Msgs:          []json.RawMessage{json.RawMessage(signedMsg.GetSignBytes())},
+		Sequence:      0,
+	}
 
+	// Create the signature bytes
+	signedBytes := sdk.MustSortJSON(keeper.Cdc.MustMarshalJSON(stdSignDoc))
 	sig, _ := base64.StdEncoding.DecodeString(msg.Signature)
 
-	if !pubkey.VerifyBytes([]byte(message), sig) {
-		return sdk.ErrUnauthorized("The session signature is not correct. " + message).Result()
+	// Verify the signature
+	if !pubkey.VerifyBytes(signedBytes, sig) {
+		return sdk.ErrUnauthorized("The session signature is not valid").Result()
 	}
 
+	// Create the testSession
 	session := types.Session{
-		SessionID:     keeper.GetLastSessionId(ctx).Next(),
-		Created:       msg.Created,
-		Expiry:        msg.Created.Add(time.Minute * 14400),
+		SessionID:     keeper.GetLastSessionID(ctx).Next(),
+		Created:       ctx.BlockHeight(),
+		Expiry:        ctx.BlockHeight() + 240, // 24 hours, counting a 6 secs block interval
```

Comment:
May put this in genesis params later.

Reply:
Opened #38 to have a better discussion there,